main.go

```diff
@@ -43,21 +44,36 @@ type CLI struct {
 
 func (cli *CLI) build(args ...string) {
 	api := api.New(cli.config, ctx)
-	build_workers := len(cli.config.Extensions)
+
+	var wg sync.WaitGroup
 	build_chan := make(chan build.Result)
 
+	errors := 0
 	for _, e := range cli.config.Extensions {
 		b := build.NewBuilder(e)
-		log.Printf("Building %s, id: %s", e.Type, e.UUID)
 
-		go b.Build(ctx, func(result build.Result) {
+		wg.Add(1)
+		go b.Build(ctx, &wg, func(result build.Result) {
```

Comment:
Does the Builder need to know about the wait group?  You could call `wg.Done()` in the callback function:

```go
func(result build.Result) {
  defer wg.Done()
  // remaining implementation
}
```

This way the waitgroup is an implementation detail of the `main` package and does not leak into `build`.